File: delivery/api/src/main/java/org/example/delivery/api/exceptionhandler/GlobalExceptionHandler.java
package org.example.delivery.api.exceptionhandler;

import lombok.extern.slf4j.Slf4j;
import org.example.delivery.api.common.api.Api;
import org.example.delivery.api.common.error.CommonErrorCode;
import org.springframework.core.annotation.Order;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.ExceptionHandler;
import org.springframework.web.bind.annotation.RestControllerAdvice;

@Slf4j
@RestControllerAdvice
@Order(Integer.MAX_VALUE)
public class GlobalExceptionHandler {

    @ExceptionHandler(Exception.class)
    public ResponseEntity<Api<?>> exception(Exception ex) {
        log.error("", ex);

        return ResponseEntity
                .status(500)
                .body(Api.error(CommonErrorCode.SERVER_ERROR));
    }
}

File: spring/issue-service/issue/src/main/kotlin/org/example/issueservice/exception/GlobalExceptionHandler.kt
package org.example.issueservice.exception

import io.github.oshai.kotlinlogging.KotlinLogging
import org.springframework.web.bind.annotation.ExceptionHandler
import org.springframework.web.bind.annotation.RestControllerAdvice

@RestControllerAdvice
class GlobalExceptionHandler {

    private val log = KotlinLogging.logger { }

    @ExceptionHandler(ServerException::class)
    fun handleServerException(ex: ServerException): ErrorResponse {
        log.error { ex.message }

        return ErrorResponse(code = ex.code, message = ex.message)
    }

    @ExceptionHandler(Exception::class)
    fun handleException(ex: Exception): ErrorResponse {
        log.error { ex.message }

        return ErrorResponse(code = 500, "Internal Server Error")
    }
}